src/components/filters/CheckBoxLargeScrn.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+import { Checkbox } from '@/components/ui/checkbox';
+import { FilterState, useFilter } from '@/context/FilterContext';
+
+interface CheckBoxLargeScrnProps {
+    name: keyof FilterState;
+    title: string;
+    options: string[];
+}
+
+export const CheckBoxLargeScrn: React.FC<CheckBoxLargeScrnProps> = ({
+    name,
+    title,
+    options,
+}) => {
+    const { selectedFilters, handleImmediateChange } = useFilter();
+
+    const selectedValues = selectedFilters[name] as string[];
+
+    const handleCheckboxChange = (value: string, checked: boolean) => {
+        const updatedValues = checked
+            ? [...selectedValues, value] // Add value if checked
+            : selectedValues.filter((v) => v !== value); // Remove value if unchecked
+
+        handleImmediateChange(name, updatedValues);
+    };
+
+    return (
+        <div className='space-y-4'>
+            <h3 className='mb-3 font-medium text-gray-750'>{title}</h3>
+            {options.map((option) => (
+                <div
+                    key={option}
+                    className='flex flex-row items-center space-x-2 align-middle text-base'
+                >
+                    <Checkbox
```

Comment:
It's a very well-structured component that renders checkbox groups on large devices, it follows best practices and is very readable.

🟡 I suggest either wrapping the <Checkbox> in the <label> tag or adding an id prop to the checkbox (or both), to improve the UX. This way clicking on the label checks the associated checkbox and adding an id prop to the checkbox.

like this:
```jsx
<label htmlFor={option} className='flex items-center space-x-2 hover:cursor-pointer'>
  <Checkbox
    id={option}
    checked={selectedValues.includes(option)}
    onCheckedChange={(checked: boolean) => {
    handleCheckboxChange(option, checked);
    }}
   />
  <span className='font-light'>{option}</span> //span element optional
</label>
```